src/io/iohk/decco/PartialCodec.scala

```diff
@@ -0,0 +1,82 @@
+package io.iohk.decco
+
+import java.nio.ByteBuffer
+
+import scala.reflect.runtime.universe.TypeTag
+
+trait PartialCodec[T] { self =>
```

Comment:
I feel it is likely that we'll get back here after using the library just to make the method covariant.

Also, it is likely that this needs documentation, I can't say what exactly a partial codec is.